ipaserver/install/ipa_idrange_fix.py

```diff
@@ -0,0 +1,1073 @@
+                    logger.info("%s", identity)
+            else:
+                logger.info("No IDs under 1000 found.")
+
+        if len(self.propositions_rid) > 0:
+            logger.info("Proposed changes to existing ranges:")
+            for id_range in self.propositions_rid:
+                logger.info(
+                    "Range '%s' - base RID: %s, secondary base RID: %s",
+                    id_range.name,
+                    id_range.base_rid,
+                    id_range.secondary_base_rid,
+                )
+        else:
+            logger.info("No changes proposed for existing ranges.")
+
+        if len(self.propositions_new) > 0:
+            logger.info("Proposed new ranges:")
+            for id_range in self.propositions_new:
+                logger.info("%s", id_range)
+        else:
+            logger.info("No new ranges proposed.")
+
+# Working with output
+# region
+
+
+def draw_ascii_table(id_ranges: list[IDRange]) -> None:
+    """Function to draw a table with ID ranges in ASCII"""
+    # Calculate the maximum width required for each column using column names
+    max_widths = {
+        column: max(
+            len(str(column)),
+            max(
+                (
+                    len(str(getattr(id_range, column)))
+                    if getattr(id_range, column) is not None
+                    else 0
+                )
+                for id_range in id_ranges
+            ),
+        )
+        for column in [
+            "name",
+            "type",
+            "size",
+            "first_id",
+            "last_id",
+            "base_rid",
+            "last_base_rid",
+            "secondary_base_rid",
+            "last_secondary_rid",
+        ]
+    }
+
+    # Draw the table header
+    header = "| "
+    for column, width in max_widths.items():
+        header += f"{column.ljust(width)} | "
+    horizontal_line = "-" * (len(header) - 1)
+    print(horizontal_line)
+    print(header)
+    print(horizontal_line)
+
+    # Draw the table rows
+    for id_range in id_ranges:
+        row = "| "
+        for column, width in max_widths.items():
+            value = getattr(id_range, column)
+            if value is not None:
+                row += f"{str(value).rjust(width)} | "
+            else:
+                # Adding the separator
+                row += " " * (width + 1) + "| "
+        print(row)
+    print(horizontal_line)
+# endregion
+# Reading from LDAP
+# region
+
+
+def read_ranges(suffix) -> list[IDRange]:
+    """Function to read ID ranges from LDAP"""
+    try:
+        ranges = api.Backend.ldap2.get_entries(
+            DN(api.env.container_ranges, suffix),
+            ldap.SCOPE_ONELEVEL,
+            "(objectclass=ipaIDRange)",
+        )
+    except errors.NotFound:
+        logger.error("LDAPError: No ranges found!")
+    except errors.ExecutionError as e:
+        logger.error("Exception while reading users: %s", e)
+    id_ranges: IDRange = []
```

Comment:
Is it worth proceeding if either of these exceptions are raised?

Reply:
fixed, added `else:` for logic